diff --git a/index.js b/index.js
index a680d25dfa98a45a3efec0bfb493500e6e9faa91..ed3cbd14483c24c622f86a53ef576be12a5a8d25 100644
--- a/index.js
+++ b/index.js
@@ -4,7 +4,15 @@ exports.run = run;
 const runTsc_1 = require("@volar/typescript/lib/quickstart/runTsc");
 const vue = require("@vue/language-core");
 const windowsPathReg = /\\/g;
-function run(tscPath = require.resolve('typescript/lib/tsc')) {
+/**
+ * ⚠ Replaced tsc with _tsc to support patched/morphed TypeScript (e.g. ts-patch).
+ * This may break in future TypeScript versions.
+ * Fallback to tsc if _tsc does not exist.
+ */
+function run(tscPath) {
+    try {
+        tscPath = require.resolve('typescript/lib/_tsc');
+    } catch {
+        tscPath = require.resolve('typescript/lib/tsc');
+    }
